arm9/src/scrcmd_party.c

```diff
@@ -0,0 +1,813 @@
+    u16* ret_ptr = GetVarPointer(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 required_nature = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    u8 party_count = (u8)GetPartyCount(SavArray_PlayerParty_get(sav_ptr->saveBlock2));
+
+    u8 i;
+    for (i = 0, *ret_ptr = 0xFF; i < party_count; i++)
+    {
+        struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+        struct Pokemon* party_mon = GetPartyMonByIndex(party, i);
+
+        BOOL party_mon_is_egg = (BOOL)GetMonData(party_mon, MON_DATA_IS_EGG, NULL);
+        if (party_mon_is_egg == FALSE)
+        {
+            u8 party_mon_nature = GetMonNature(party_mon);
+            if (required_nature == party_mon_nature)
+            {
+                *ret_ptr = i;
+                break;
+            }
+        }
+    }
+
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_GetPartyMonFriendship(struct ScriptContext* ctx)
+{
+    struct UnkSavStruct80* sav_ptr = ctx->unk80;
+    u16* ret_ptr = GetVarPointer(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 mon_slot = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    *ret_ptr = (u16)GetMonData(party_mon, MON_DATA_FRIENDSHIP, NULL);
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_AddPartyMonFriendship(struct ScriptContext* ctx)
+{
+    struct UnkSavStruct80* sav_ptr = ctx->unk80;
+    u16 friendship_to_add = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 mon_slot = VarGet(ctx->unk80, ScriptReadHalfword(ctx));
+    u16 map_sec = MapHeader_GetMapSec(*ctx->unk80->mapId);
+    struct PlayerParty* party = SavArray_PlayerParty_get(sav_ptr->saveBlock2);
+    struct Pokemon* party_mon = GetPartyMonByIndex(party, mon_slot);
+
+    u16 friendship = (u16)GetMonData(party_mon, MON_DATA_FRIENDSHIP, NULL);
+
+    if (friendship_to_add != 0)
+    {
+        u16 party_mon_held_item = (u16)GetMonData(party_mon, MON_DATA_HELD_ITEM, NULL);
+        u32 held_item_hold_effect = GetItemAttr(party_mon_held_item, ITEMATTR_HOLD_EFFECT, 11);
+        if (held_item_hold_effect == HOLD_EFFECT_FRIENDSHIP_UP)
+        {
+            friendship_to_add = (u16)((friendship_to_add * 150) / 100);
+        }
+
+        u32 party_mon_pokeball = GetMonData(party_mon, MON_DATA_POKEBALL, NULL);
+        if (party_mon_pokeball == ITEM_LUXURY_BALL)
+        {
+            friendship_to_add++;
+        }
+
+        u32 party_mon_egg_met_location = GetMonData(party_mon, MON_DATA_EGG_MET_LOCATION, NULL);
+        if (map_sec == party_mon_egg_met_location)
+        {
+            friendship_to_add++;
+        }
+    }
+
+    friendship += friendship_to_add;
+    if (friendship > 0xFF)
+    {
+        friendship = 0xFF;
+    }
+
+    SetMonData(party_mon, MON_DATA_FRIENDSHIP, &friendship);
+
+    return FALSE;
+}
+
+THUMB_FUNC BOOL ScrCmd_DepletePartyMonFriendship(struct ScriptContext* ctx)

```

Comment:
SubtractPartyMonFriendship (or RemovePartyMonFriendship) would be preferable here imo